packages/eslint-plugin/tests/rules/adjacent-overload-signatures.test.ts

```diff
@@ -256,6 +256,24 @@ class Test {
   invalid: [
     {
       code: `
+function wrap() {
+  function foo(s: string);
+  function foo(n: number);
+  export type bar = number;
+  function foo(sn: string | number) {}
+}
+      `,
+      errors: [
+        {
+          messageId: 'adjacentSignature',
+          data: { name: 'foo' },
+          line: 6,
+          column: 3,
+        },
+      ],
+    },
```

Comment:
Let's add some more testing so we can be more confident about this change:
* Some test cases in `valid` 
* At least one other kind of block statement - to make sure it's not just `function` blocks that work

Reply:
I've added two valid test cases and another invalid case. However, other than `if` statements and function declarations, I can't think of some more common scenes where people would define functions... What do you think?

Reply:
Yeah I think it's great as-is. In theory someone _could_ put them in other places like `while` loops, but IMO we don't need to exhaustively check every single possible block statement type. Just a few to make sure the rule works across a few different types.